importer/likutim/create_unit.go

```diff
@@ -0,0 +1,237 @@
+package likutim
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"github.com/volatiletech/sqlboiler/boil"
+	"github.com/volatiletech/sqlboiler/queries/qm"
+	"gopkg.in/volatiletech/null.v6"
+
+	"github.com/Bnei-Baruch/mdb/common"
+	"github.com/Bnei-Baruch/mdb/models"
+	"github.com/Bnei-Baruch/mdb/utils"
+)
+
+type CreateUnits struct {
+	duplicates []*Double
+	mdb        *sql.DB
+}
+
+func (c *CreateUnits) Run() {
+	err := c.duplicatesFromJSON()
+	if err != nil {
+		log.Errorf("Error on read. Error: %s", err)
+		compare := new(Compare)
+		compare.Run()
+		c.duplicates = compare.result
+	}
+
+	c.openDB()
+	defer c.mdb.Close()
+
+	err = os.MkdirAll(viper.GetString("likutim.results-dir"), os.ModePerm)
+	if err != nil {
+		log.Errorf("Can't create directory: %s", err)
+	}
+	c.updateUnits()
+}
+
+func (c *CreateUnits) updateUnits() {
+	for _, d := range c.duplicates {
+		log.Debugf("\n\nStart create new unit type LIKUTIM, file: %s all files: %v", d.Save, d.Doubles)
+
+		tx, err := c.mdb.Begin()
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		err, cu := c.createUnitOnTransaction(tx, d.Save)
+		if err != nil {
+			log.Error(err)
+			tx.Rollback()
+			continue
+		}
+		err = tx.Commit()
```

Comment:
better have 1 transaction for each iteration. No need transaction for the inner loop.

A nice way to depict transaction is: "All this things inside the transaction must either all succeed together or none of them at all"